### PR TITLE
YAML

### PR DESCRIPTION
**Pull request by PanosCodes**

### CODE REVIEW

File: test/stdlib/YAML_test.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  library 'yaml'
+  testing 'singleton(::YAML)'
+
+  def test_load_returns_a_string
+    assert_send_type  "(String) -> (Hash | String | Array)", YAML, :load, "abcde"
+  end
+
+  def test_load_returns_a_hash
+    assert_send_type  "(String) -> (Hash | String | Array)", YAML, :load, "--- \n- Apple \n- Orange "
+  end
```
Comment (PanosCodes) [inline test/stdlib/YAML_test.rb:15]: Running `bundle exec rake test TEST=./test/stdlib/YAML_test.rb` 

Results in 
```
Minitest::Assertion: Call trace does not match with given method type: #<struct RBS::Test::CallTrace method_name=:load, method_call=#<RBS::Test::ArgumentsReturn:0x00007ff9856131c8 @arguments=["--- \n- Apple \n- Orange "], @exit_value=["Apple", "Orange"], @exit_type=:return>, block_calls=[], block_given=false>.
Expected ["[Moduleload] ReturnTypeError: expected `Hash | String | Array` but returns `[\"Apple\", \"Orange\"]`"] to be empty.
```

Is there something wrong with the definition of the return signature ?

Comment (PanosCodes) [reply test/stdlib/YAML_test.rb]: Arrays should be declared `Array[untyped]` and Hash `Hash[untyped, untyped]`

File: stdlib/yaml/store.rbs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  #
+  def initialize: (*untyped o) -> YAML::Store
+
+  def dump: (untyped table) -> String
+
+  def empty_marshal_checksum: () -> String
+
+  def empty_marshal_data: () -> String
+
+  def load: (String) -> (Hash | String | Array)
```
Comment (soutaro) [inline stdlib/yaml/store.rbs:50]: `Hash[untyped, untyped] | String | Array[untyped]` ??
Or I think returning just `untyped` is enough.

File: stdlib/yaml/dbm.rbs [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    # Converts the contents of the database to an array of [key, value] arrays,
+    # and returns it.
+    def to_a: () -> Array [Array] 
+
+    # :call-seq:
+    #   ydbm.to_hash -> hash
+    #
+    # Converts the contents of the database to an in-memory Hash object, and
+    # returns it.
+    def to_hash: () -> Hash
```
Comment (soutaro) [inline stdlib/yaml/dbm.rbs:219]: `Hash[untyped, untyped]`

**Comment by soutaro** [comment]
Hi!

Rewriting `Hash` to `Hash[untyped, untyped]` would make the test pass.
However, I'm not very sure if returning hash types makes much sense, because this is serialization and `untyped` should appear almost everywhere...

**Comment by PanosCodes** [comment]
> Hi!
> 
> Rewriting `Hash` to `Hash[untyped, untyped]` would make the test pass.
> However, I'm not very sure if returning hash types makes much sense, because this is serialization and `untyped` should appear almost everywhere...

Thank you for your feedback. 
So you suggest when a there are unknown types inside the hash instead of `Hash[untyped, untyped]` we should return `untyped` ?

**Comment by soutaro** [comment]
Oh no, I'm sorry. It is about `DBM`. I was confusing a bit about if it was about YAML.
`Hash[untyped, untyped]` would make sense for method which returns a hash.

**Comment by PanosCodes** [comment]
```
/__w/rbs/rbs/lib/rbs/errors.rb:140:in `check!': /__w/rbs/rbs/stdlib/yaml/store.rbs:32:0...53:3: Could not find super class: PStore (RBS::NoSuperclassFoundError)
```

The YAML Store depends on PStore for witch we don't have signature. That is probably the reason the tests fail.
Here is a PR from PStore https://github.com/ruby/rbs/pull/443

**Comment by PanosCodes** [comment]
```
/__w/rbs/rbs/lib/rbs/errors.rb:140:in `check!': /__w/rbs/rbs/stdlib/yaml/dbm.rbs:61:2...220:5: Could not find super class: ::DBM (RBS::NoSuperclassFoundError)
```
YAML also depends on DBM and tries to find it.
Here is PR for that https://github.com/ruby/rbs/pull/441

**Comment by PanosCodes** [comment]
@soutaro Any idea why it fails to find DBM ? I believe https://github.com/ruby/rbs/pull/441 should have fixed that.

```
/__w/rbs/rbs/lib/rbs/errors.rb:140:in `check!': /__w/rbs/rbs/stdlib/yaml/dbm.rbs:61:2...220:5: Could not find super class: ::DBM (RBS::NoSuperclassFoundError)
```

**Comment by soutaro** [comment]
@PanosCodes Could you rebase your branch on the latest `HEAD` and moving your RBS files under `stdlib/yaml/0` directory?

Standard library signatures are moved in #405. (I'm sorry for breaking your work with that. 🙇‍♂️)

**Comment by PanosCodes** [comment]
> @PanosCodes Could you rebase your branch on the latest `HEAD` and moving your RBS files under `stdlib/yaml/0` directory?
> 
> Standard library signatures are moved in #405. (I'm sorry for breaking your work with that. 🙇‍♂️)

No worries 😄 But it still throws the same error 🤔

**Comment by soutaro** [comment]
@PanosCodes I got it. 💪 

You need to specify `-r dbm -r pstore` to command line to start the validation.

```
# You can add the following lines to Rakefile to let `rake validate` run for YAML.

if lib == ["yaml"]
  lib << "dbm"
  lib << "pstore"
end
```

**Comment by PanosCodes** [comment]
> @PanosCodes I got it. 💪
> 
> You need to specify `-r dbm -r pstore` to command line to start the validation.
> 
> ```
> $ bundle exec rbs -r yaml -r dbm -r pstore validate
> ```
> 
> ```ruby
> # You can add the following lines to Rakefile to let `rake validate` run for YAML.
> 
> if lib == ["yaml"]
>   lib << "dbm"
>   lib << "pstore"
> end
> ```

Awesome , thank you for looking into that  👍

**Review by soutaro — APPROVED** [review]
Thanks! @PanosCodes

**Pull request merged**